### PR TITLE
AVSW-51318 port elision code from statsite

### PR DESCRIPTION
**Pull request by dgrala**

### CODE REVIEW

**Comment by dgrala** [comment]
PTAL @golfdish

File: src/stats.c [hunk 1]
```diff
@@ -651,6 +651,9 @@ stats_server_t *stats_server_create(struct ev_loop *loop,
     server->validator = validator;
     server->validate_point_tags = NULL;
 
+    elide_destroy(server->elide);
+    server->elide = NULL;
+
```
[the remark this answers is not in the document]
Comment (dgrala) [reply src/stats.c]: Good catch! I'm not sure why it ended up here. Moved it to `stats_server_destroy`

**Comment by golfdish** [comment]
As I alluded to in https://github.com/lyft/k8s-sidecars/pull/2772, we don't want to merge this into master right away, but create a release from the finalized code so that we can control deployment.

**Comment by dgrala** [comment]
Passing old and new tests ```
Ran 26 tests in 49.833s

OK
```

**Comment by dgrala** [comment]
PTAL @golfdish

**Comment by golfdish** [comment]
This looks good. I'm going to recommend merging into the new `elision-release` branch in lieu of master, and we can tag that one for release. I'll modify the pull request to do that if there are no objections.

**Pull request merged**